File: tournaments/factorSum/factorSum.js
function factorSum(n) {
    let result = n;
    let a = 0;
    let b = 0;

    do {
        let c = 2;
        a = result;
        b = result;
        result = 0;

        while (b >= c**2) {
            if (!(b % c)) {
                b /= c;
                result += c;
            } else ++c;
        }

        result += b;
    } while (result !== a);

    return result;
}

File: tournaments/factorSum/factorSum.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub factorSum {
    my ($n) = @_;

    my $result = $n;
    my $c = 0;
    my $d = 0;

    do {
        my $e = 2;
        $c = $result;
        $d = $result;
        $result = 0;

        while ($d >= $e**2) {
            if (!($d % $e)) {
                $d /= $e;
                $result += $e;
            } else {
                ++$e;
            }
        }

        $result += $d;
    } until ($result == $c);

    return $result;
}
